### ISSUE TITLE
需要类似pytorch那样的gradient checkpointing功能

### ISSUE THREAD

**Issue by CSQianDong** [PaddlePaddle/Paddle#46588]
### 需求描述 Feature Description

任务目标：提升训练batch size；
需求场景：搜索训练负采样个数显著影响模型效果；
功能描述：https://blog.csdn.net/lavinia_chen007/article/details/113609838，，参考pytorch的gradient checkpointing功能，现有的recompute不支持DDP分布式训练，DP模式训练效率太低。

**Comment by yeliang2258** [comment]
https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/callbacks/ModelCheckpoint_cn.html#modelcheckpoint
您好，可以看看这个API是否满足你的使用要求

**Comment by CSQianDong** [comment]
肯定不能呀，这个是保存模型的。参考这个[pytorch-gradient checkpointing](https://blog.csdn.net/lavinia_chen007/article/details/113609838%EF%BC%8C%EF%BC%8C%E5%8F%82%E8%80%83pytorch%E7%9A%84gradient)

**Comment by ForFishes** [comment]
您好，可以使用paddle的recompute接口，这个功能使用文档：https://www.paddlepaddle.org.cn/documentation/docs/zh/api/paddle/distributed/fleet/utils/recompute_cn.html#recompute。
当使用recompute接口，同时使用DataParallel情况下，需要使用no_sync+fused_allreduce实现梯度同步。
文档：https://www.paddlepaddle.org.cn/documentation/docs/zh/api/paddle/DataParallel_cn.html#id1

**Comment by CSQianDong** [comment]
我了解这个用法，我的需求是想让paddle支持DDP模式的recompute，DP模式需要同步模型参数到卡0，然后再分发给各个卡，存在效率低的问题
